Updated Tracker project to target .Net Standard 2.0

fahadabdulaziz opened the pull request:
Updated Piwik.Tracker Project to target .NET Standard 2.0.
Updated other projects to .NET Framework 4.7 as we should keep back compatibility with least 3 years old technologies.
Test Units did not run at all with me, did not check as we need to upgrade them to xUnit.

I think we should move samples to other repo so could later convert them to templates and comply with how .NET core sdk works now.

@mattab

fahadabdulaziz committed:
Updated Tracker project to target .Net Standard 2.0

fahadabdulaziz committed:
Update .travis.yml

Remove usage of mono to use .NET core for running tests

fahadabdulaziz committed:
Update .travis.yml

Changed .NET version

fahadabdulaziz committed:
Update .travis.yml

Fix folder name

mattab commented:
Thanks for creating the PR. 
@julienmoumne @ptr1120 if you have any feedback about this PR looking at the code, feedback welcome! 

@fahadabdulaziz I give you already commit access but please wait maybe 1 week or so before merging this first one :-)

mattab commented:
fyi @julienmoumne @ptr1120  `fahadabdulaziz` has been added as a collaborator with write permissions on the repository.

fahadabdulaziz commented:
Thank you @mattab 
By the way, the tests didn't run on travis.

mattab commented:
@fahadabdulaziz thanks for pointing it out. when a test fails, the PR check should appear red, but it is green now :thinking: 
Maybe you are able to look into this issue?

fahadabdulaziz commented:
@mattab The CI was using Mono to build & test the project, now when we convert the project to .NET standard we use .NET Core for that. but test not working neither on my machine and the CI.

Anyway I'll change this PR to be vNext of the project (Including rewrite the tests and make it work)

SpoinkyNL commented:
Hello,

I was wondering if there will be any further updates to this? 

Thanks!

QuinnDamerell commented:
Any updates on this? It's 2025 and this package only supports .net framework 4.5 :D

fahadabdulaziz committed:
Refactor project to .NET 8.0 and improve HTTP handling

Updated `Piwik.Tracker.Tests.csproj` to SDK-style format targeting .NET 8.0, adding necessary package references for testing.

Replaced `JavaScriptSerializer` with `JsonConvert` in `PiwikTrackerTests.cs` for JSON serialization and improved query string parsing.

Refactored `PiwikTrackerWithMockedServerTests.cs` to use `Mock<HttpMessageHandler>` for better HTTP client testing.

Removed `AssemblyInfo.cs` as it's no longer needed in the new project format and simplified `app.config` by removing outdated binding redirects.

Updated package references in `Piwik.Tracker.csproj` for `Microsoft.AspNetCore.Http` and `Newtonsoft.Json` to their latest versions.

Significantly refactored `PiwikTracker.cs` to utilize `HttpClient` for HTTP requests, enhancing dependency injection and overall performance.
